base/show.jl

```diff
@@ -79,11 +79,21 @@ function show(io::IO, ::MIME"text/plain", t::AbstractDict{K,V}) where {K,V}
     if !haskey(io, :compact)
         recur_io = IOContext(recur_io, :compact => true)
     end
+    sorted = get(io, :sorted, false)
 
     summary(io, t)
     isempty(t) && return
     print(io, ":")
     show_circular(io, t) && return
+
+    if sorted
+        try # sorting fails when elements are not comparable, collect can fail too
+            t = sort!(collect(t))
```

Comment:
You can have a lazy dict type that maps all `typemin(Int):typemax(Int)` to something. Materializing this as a vector is quite dangerous.

Reply:
Good point. Maybe the `:sorted` attribute should get an `Int` instead of a `Bool`, serving as a threshold for the maximum size of the dict below which it's sorted.

Reply:
Another possibility is a dictionary variant of https://github.com/JuliaArrays/MappedArrays.jl where the mapping function is super slow. It is perhaps better to use a variant of `Iterators.take` that takes a timeout.

Reply:
Again good point! I'm starting to wonder whether this is really feasable in the general case...